### PR TITLE
Add setting for Support Infill Line Direction

### PR DESCRIPTION
**Pull request by fieldOfView**
This PR makes the orientation of infill lines settable. It fixes https://github.com/Ultimaker/Cura/issues/3481

Also see the matching PR for the engine: https://github.com/Ultimaker/CuraEngine/pull/819

### CODE REVIEW

**Comment by fieldOfView** [comment]
Fixed the link in the PR. Can't fix the one that was sent out in the notifications.

**Comment by smartavionics** [comment]
What I mean is that your commit contains the following description which doesn't match my idea of what the setting is doing:

````
"description": "The minimum angle of overhangs for which support is added. At a value of 0° all overhangs are supported, 90° will not provide any support.",
````

**Comment by fieldOfView** [comment]
Ah, right, I’ll fix that.

**Comment by diegopradogesto** [comment]
Together with the PR in CuraEngine: CURA-5635

File: resources/definitions/fdmprinter.def.json [hunk 1]
```diff
@@ -3873,6 +3873,19 @@
...
+                "support_infill_angle":
+                {
+                    "label": "Support Infill Line Direction",
+                    "description": "Orientation of the infill pattern for supports. The support infill pattern is rotated in the horizontal plane.",
+                    "unit": "°",
+                    "type": "float",
+                    "minimum_value": "-90",
+                    "maximum_value": "90",
+                    "default_value": 0,
+                    "enabled": "support_enable",
```
Comment (alekseisasin) [inline resources/definitions/fdmprinter.def.json]: Could you please disable the setting if _Support Pattern_ is Concentric or Cross support.

Comment (fieldOfView) [reply resources/definitions/fdmprinter.def.json]: Good catch. I've also disabled the setting if the Support Density is set to 0 (in analogy to Infill Line Directions.

**Comment by fieldOfView** [comment]
I have widened the range of allowed angles (from -90, 90 to -180,180). Even though the difference is very subtle with the current patterns, there is a different (with zigzag) and with future patterns the difference may be more dramatic.

**Pull request merged**